src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.StateManager.HostStates.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.EngineV1
+{
+    internal partial class DiagnosticIncrementalAnalyzer
+    {
+        private partial class StateManager
+        {
+            /// <summary>
+            /// This class is responsible for anything related to <see cref="DiagnosticState"/> for host level <see cref="DiagnosticAnalyzer"/>s.
+            /// </summary>
+            private class HostStates
+            {
+                private readonly StateManager _owner;
+
+                private ImmutableDictionary<string, ImmutableDictionary<DiagnosticAnalyzer, StateSet>> _stateMap;
```

Comment:
Ah well, there you go. No more Provider IDs, seems to simplify everything quite a bit!!